src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,802 @@
+            telemetry.set_exception(exception=e, fault_type=consts.Storage_Available_Fault_Type, summary="Error while trying to create diagnostic logs folder")
+            cli_outputs.append("An expection has occured while creating the diagnostic logs folder in your local machine. Exception: {}".format(str(e)) + "\n")
+            return "", "folder_not_created"
+
+    except Exception as e:
+        logger.warning("An expection has occured while creating the diagnostic logs folder in your local machine. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Storage_Available_Fault_Type, summary="Error while trying to create diagnostic logs folder")
+        cli_outputs.append("An expection has occured while creating the diagnostic logs folder in your local machine. Exception: {}".format(str(e)) + "\n")
+        return "", "folder_not_created"
+
+
+def connected_cluster_logger(filepath_with_timestamp, connected_cluster, storage_space_available):
+
+    global cli_outputs
+
+    try:
+        connected_cluster_resource_path = os.path.join(filepath_with_timestamp, "Connected_cluster_resource.txt")
+        if storage_space_available:
+            with open(connected_cluster_resource_path, 'w+') as cc:
+                cc.write(str(connected_cluster))
+        return "Passed", storage_space_available
+
+    except OSError as e:
+        if "[Errno 28]" in str(e):
+            storage_space_available = False
+            telemetry.set_exception(exception=e, fault_type=consts.No_Storage_Space_Available_Fault_Type, summary="No space left on device")
+            shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+        else:
+            logger.warning("An expection has occured while trying to store the connected cluster resource logs from the cluster. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Connected_Cluster_Resource_Fault_Type, summary="Eror occure while storing the connected cluster resource logs")
+            cli_outputs.append("An expection has occured while trying to store the connected cluster resource logs from the cluster. Exception: {}".format(str(e)) + "\n")
+
+    except Exception as e:
+        logger.warning("An expection has occured while trying to store the connected cluster resource logs from the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Connected_Cluster_Resource_Fault_Type, summary="Eror occure while storing the connected cluster resource logs")
+        cli_outputs.append("An expection has occured while trying to store the connected cluster resource logs from the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return "Failed", storage_space_available
+
+
+def arc_agents_logger(corev1_api_instance, filepath_with_timestamp, storage_space_available):
+
+    global cli_outputs
```

Comment:
let's rename to something more relevant - may be diagnoser_results

Reply:
Changed it to diagnoser_results.